### PR TITLE
Added support for radio-row

### PR DESCRIPTION
**Pull request by Ibrahimhass**
[Component Screenshot](https://drive.usercontent.google.com/u/0/uc?id=1K09WWxueQ4Ffh30M5lGWs40aPqWWDu4O)

### CODE REVIEW

File: lib/components/radio_row/radio_row.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          const SizedBox(
+            width: 20.0,
+          ),
+          Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(
+                title,
+                style: theme.typography.heading7
+                    .copyWith(fontWeight: FontWeight.w600),
```
Comment (kalpeshp0310) [inline lib/components/radio_row/radio_row.dart:62]: I guess, You are also confused by Figma's UI just like me a few months back.
Figma shows all the properties of the style for on component along with their style token name. 

Style token size: 
Screenshot 2024-04-07 at 2 16 27 PM


Style size in component:
Screenshot 2024-04-07 at 2 16 54 PM

You can see font-weight is the same as what was given to the base token `headline7`.
We just need to specify `headline7` as the font style and do not need to override font-weight. 
Hope Figma makes their UI clear in the future 🤞

File: lib/components/radio_row/radio_row.dart [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+              ),
+              if (showDescription)
+                Text(
+                  description,
+                  style: theme.typography.body7.copyWith(
+                    fontWeight: FontWeight.w400,
```
Comment (kalpeshp0310) [inline lib/components/radio_row/radio_row.dart:68]: Same here.

File: lib/components/radio_row/radio_row.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:tarka_ui/styles/theme.dart';
+
+import '../radio_button/radio_button.dart';
+
+class TUIRadioRow extends StatelessWidget {
+  final String title;
+  final String description;
+  final VoidCallback? onPressed;
```
[the remark this answers is not in the document]
Comment (Ibrahimhass) [reply lib/components/radio_row/radio_row.dart:9]: No it is being used, please check.
```
  TUIRadioButton getRadioButton() {
    if (isSelected == null) {
      return TUIRadioButton(
        onPressed: () {},
      );
    } else {
      return TUIRadioButton(
        onPressed: () {},
        isSelected: isSelected ?? false,
      );
    }
  }

```

File: lib/components/radio_row/radio_row.dart [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  final bool? isSelected;
+  final bool backgroundDark;
+
+  const TUIRadioRow({
+    super.key,
+    required this.title,
+    this.description = "",
```
[the remark this answers is not in the document]
Comment (Ibrahimhass) [reply lib/components/radio_row/radio_row.dart:16]: I am following the API desgin similar to Checkbox Row.
Also it is more convenient to have a simple check like 
`final showDescription = description.isNotEmpty;`

File: lib/components/radio_row/radio_row.dart [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:tarka_ui/styles/theme.dart';
+
+import '../radio_button/radio_button.dart';
+
+class TUIRadioRow extends StatelessWidget {
+  final String title;
+  final String description;
+  final VoidCallback? onPressed;
+  final bool? isSelected;
```
[the remark this answers is not in the document]
Comment (Ibrahimhass) [reply lib/components/radio_row/radio_row.dart:10]: Thanks for the suggestion, I have made this non-nullable now.

File: lib/components/radio_row/radio_row.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ibrahimhass) [reply lib/components/radio_row/radio_row.dart:9]: Fixed this now.

File: lib/components/radio_row/radio_row.dart (hunk 1 above)
Comment (Ibrahimhass) [reply lib/components/radio_row/radio_row.dart:62]: Awesome thanks for sharing the observation :).

File: lib/components/radio_row/radio_row.dart (hunk 2 above)
Comment (Ibrahimhass) [reply lib/components/radio_row/radio_row.dart:68]: Fixed

**Review by kalpeshp0310 — APPROVED** [review]

**Pull request merged**